packs/st2/actions/upload_to_s3.py

```diff
@@ -1,21 +1,51 @@
+import os
+import httplib
+
+import requests
+
 from lib.action import St2BaseAction
-import s3
 
 __all__ = [
     'UploadToS3'
 ]
 
+S3_BUCKET_URL = 'https://%(bucket)s.s3.amazonaws.com/'
+
 
 class UploadToS3(St2BaseAction):
-    def run(self, file_name, remote_file, bucket):
-        remote_name = s3.S3Name(remote_file, bucket=bucket)
-
-        connection = s3.S3Connection(**self.config['s3'])
-        storage = s3.Storage(connection)
-        storage.write(file_name, remote_name)
-        payload = {
-            "status": "ok",
-            "uploaded_file": remote_file,
+    """
+    Note: The destination bucket me be configured so everyone can write to it,
+    but only we can read from it.
+    """
+
+    def run(self, bucket, file_name, remote_file=None):
+        if not remote_file:
+            remote_file = os.path.basename(remote_file)
+
+        if not os.path.isfile(file_name):
+            raise ValueError('Local file "%s" doesn\'t exist' % (file_name))
+
+        url = S3_BUCKET_URL % {'bucket': bucket}
+        url = url + remote_file
+
+        # Note: Requests library performs streaming and chunked upload
+        files = {'file': open(file_name, 'rb')}
+        response = requests.put(url=url, files=files, verify=False)
```

Comment:
There is no need to use S3 library here since it's a simple PUT http request.

I also removed verify=False, I used it for debugging purposes.